source/isaaclab/isaaclab/assets/asset_base.py

```diff
@@ -254,17 +255,23 @@ def _initialize_callback(self, event):
             PhysX handles are only enabled once the simulator starts playing. Hence, this function needs to be
             called whenever the simulator "plays" from a "stop" state.
         """
-        if not self._is_initialized:
-            # obtain simulation related information
-            sim = sim_utils.SimulationContext.instance()
-            if sim is None:
-                raise RuntimeError("SimulationContext is not initialized! Please initialize SimulationContext first.")
-            self._backend = sim.backend
-            self._device = sim.device
-            # initialize the asset
-            self._initialize_impl()
-            # set flag
-            self._is_initialized = True
+        try:
+            if not self._is_initialized:
+                # obtain simulation related information
+                sim = sim_utils.SimulationContext.instance()
+                if sim is None:
+                    raise RuntimeError(
+                        "SimulationContext is not initialized! Please initialize SimulationContext first."
+                    )
+                self._backend = sim.backend
+                self._device = sim.device
+                # initialize the asset
+                self._initialize_impl()
```

Comment:
Maybe do a try catch only over this call instead of the whole function? Don't see the need to wrap around the whole thing.